### PR TITLE
fpga: finalize plugin kustomization

### LINKED ISSUE #318: Finalize FPGA kustomize
- [ ] Move FPGA kustomization out of “beta” and align with changes in #301.
- [ ] Update documentation
- [ ] Update the screencast to match kustomization based deployment
- [ ] Add kustomization for `opae-nlb-demo` deployments

### PR DESCRIPTION
**Pull request by rojkov**
closes #318

### CODE REVIEW

File: cmd/fpga_plugin/README.md [hunk 1]
```diff
@@ -177,82 +172,83 @@ $ ls /var/lib/kubelet/device-plugins/kubelet.sock
...
+```
 
-This deployment model is under development. The remaining part of this
-document goes through the current deployment model: here for the
-FPGA plugin and in the next document for the FPGA admission controller.
+If you intend to deploy your own image, you will need to reference the
+[image build section](#build-the-plugin-image) first.
 
-### Create a service account
+If you do not want to deploy the `devel` tagged image, you will need to create your
```
Comment (mythi) [inline cmd/fpga_plugin/README.md:233]: `devel` or release tagged image...

**Comment by bart0sh** [comment]
scripts/deploy-fpgaplugin.sh should be removed and FPGA screencasts should be updated I guess. It would be good to have these changes in the same PR. I can take care of screencasts if needed.

**Comment by rojkov** [comment]
Ed, I've updated the screencast script. Could you please try to run it? And the E2E test too?

File: cmd/fpga_plugin/README.md (hunk 1 above)
Comment (rojkov) [reply cmd/fpga_plugin/README.md]: fixed

**Comment by bart0sh** [comment]
> should we do all existing and pending screencasts updates after or before this PR is merged?

Should we drop nlb3 orchestrated demo in favour of [OpenCL demo](https://github.com/intel/intel-device-plugins-for-kubernetes/pull/410)?

File: cmd/fpga_plugin/README.md [hunk 2]
```diff
@@ -177,82 +172,83 @@ $ ls /var/lib/kubelet/device-plugins/kubelet.sock
...
+```
 
-This deployment model is under development. The remaining part of this
-document goes through the current deployment model: here for the
-FPGA plugin and in the next document for the FPGA admission controller.
+If you intend to deploy your own image, you will need to reference the
+[image build section](#build-the-plugin-image) first.
 
-### Create a service account
+If you do not want to deploy the `devel` or `release` tagged image, you will need to create your
```
Comment (mythi) [inline cmd/fpga_plugin/README.md:233]: I meant to say a tag with the release version (e.g., `0.17.0`...)

Comment (rojkov) [reply cmd/fpga_plugin/README.md]: Right, that was a mindless fix. Removed the backticks.

**Comment by kad** [comment]
> > should we do all existing and pending screencasts updates after or before this PR is merged?
> 
> Should we drop nlb3 orchestrated demo in favour of [OpenCL demo](https://github.com/intel/intel-device-plugins-for-kubernetes/pull/410)?

I would keep it.  I mean, update to have demos for both OpenCL and GBS in both modes.

**Comment by bart0sh** [comment]
@mythi, @kad, @rojkov should we do all existing and pending screencasts updates after or before this PR is merged?

**Comment by rojkov** [comment]
@bart0sh I've tried to adapt your latest screencast too. Hopefully it works now, I haven't tested it.

**Comment by bart0sh** [comment]
@rojkov tested preprogrammed scenario. Screencast script works with these small changes:
```
diff --git a/demo/screencast-fpga-preprogrammed.sh b/demo/screencast-fpga-preprogrammed.sh
index 144dccf..102a4d8 100755
--- a/demo/screencast-fpga-preprogrammed.sh
+++ b/demo/screencast-fpga-preprogrammed.sh
@@ -80,7 +80,7 @@ screen3()
 {
   clear
   out '3. Deploy FPGA plugin'
-  command 'kubectl apply -k deployments/fpga_plugin/overlays/af'
+  command 'kubectl apply -k plugins/deployments/fpga_plugin/overlays/af'
   sleep 3
   out 'Deploy example mappings:'
   command 'kubectl apply -f plugins/deployments/fpga_admissionwebhook/mappings-collection.yaml'
@@ -101,7 +101,7 @@ screen3()
 screen4()
 {
   clear
-  out '5. Run OPAE workload that uses NLB3 bitstream'
+  out '4. Run OPAE workload that uses NLB3 bitstream'
   out 'Check if devices are programmed with NLB3:'
   command 'cat /sys/class/*/*/*/afu_id'
   out 'Run workload:'
```

please, apply

**Comment by rojkov** [comment]
Thanks! Applied the patch.

**Pull request merged**